### PR TITLE
Implement AMQP broker

### PR DESCRIPTION
**Pull request by mathewpower**

### CODE REVIEW

File: drivers/amqp/broker.go [hunk 1]
```diff
@@ -1,23 +1,131 @@
...
 }
 
 func newAMQP(uri string) (celeriac.Broker, error) {
-	fmt.Println("Loading AMQP driver...")
+	log.Println("Loading AMQP driver...")
+	log.Println("Connecting to Broker!")
+	connection, err := amqp.Dial(uri)
+
+	if err != nil {
+		log.Fatal("Unable to connect to AMQP broker")
```
Comment (justcompile) [inline drivers/amqp/broker.go:125]: I would log as an error and then return the error rather than bombing out here

File: drivers/amqp/broker.go [hunk 2]
```diff
@@ -1,23 +1,131 @@
...
 // AMQP is an instance of a celeriac Broker which uses amqp
-type AMQP struct{}
+type AMQP struct{
+	Connection amqp.Connection
+}
 
 // Close closes the connection to the broker
 func (a *AMQP) Close() error {
-	return nil
+	err := a.Connection.Close()
```
Comment (justcompile) [inline drivers/amqp/broker.go:19]: I'd just return `a.Connection.Close()` rather than declaring a new var

File: drivers/amqp/broker.go [hunk 3]
```diff
@@ -1,23 +1,131 @@
...
+	"github.com/google/uuid"
+	"github.com/streadway/amqp"
 
 	"github.com/go-celeriac/celeriac"
 )
 
 // AMQP is an instance of a celeriac Broker which uses amqp
-type AMQP struct{}
+type AMQP struct{
+	Connection amqp.Connection
```
Comment (justcompile) [inline drivers/amqp/broker.go:14]: Any particular reason for not using a pointer here?

**Review by justcompile — COMMENTED** [review]
Not wildly different to what I started to build out.

One question though, why are we opening a channel with each (En|de)queue function?

File: drivers/amqp/broker.go (hunk 3 above)
Comment (mathewpower) [reply drivers/amqp/broker.go:14]: No good reason

File: drivers/amqp/broker.go (hunk 1 above)
Comment (mathewpower) [reply drivers/amqp/broker.go:125]: QueueDeclare is idempotent in that it only creates the queue if one doesn't already exist. 

The reason why I did this on each (En|de)queue was to allow the user to put messages on different queues using the same connection. 

Not sure if this is good idea or the best why to do this...

**Comment by mathewpower** [comment]
Closed in favour of https://github.com/go-celeriac/celeriac/pull/4

**Pull request closed without merging**